61_UI/main.cpp

```diff
@@ -589,6 +589,10 @@ class UISampleApp final : public examples::SimpleWindowedApplication
 				auto* pipeline = pass.ui.manager->getPipeline();
 				cb->bindGraphicsPipeline(pipeline);
 				cb->bindDescriptorSets(EPBP_GRAPHICS, pipeline->getLayout(), uiParams.resources.texturesInfo.setIx, 1u, &pass.ui.descriptorSet.get()); // note that we use default UI pipeline layout where uiParams.resources.textures.setIx == uiParams.resources.samplers.setIx
+				
+				if (!keepRunning())
+					return;
+				
```

Comment:
ok I get the bug, its because the window close callback signal can be handled at any time, destroying your surface's swapchain and stuff, so all one needs to do it just refrain from performing a submit that uses the surface/swapchain resources and the present call.

Therefore a mutex `closeWindowMutex` needs to be put into the default Window Callback OR the Simple Managed surface which gets locked between the window close callback and the time the callback gets dropped (so we are sure window refctd pointer is dead already - Nabla dropped the window).

Then that mutex should stay locked (and `irrecoverable()` needs to be checked before lock) around the submit and present calls, or both at the same time if they are close together.

Reply:
we will assign someone to fix stuff & test it in another PR